### PR TITLE
Fix rooted slot iterator

### PR DESCRIPTION
**Pull request by carllin**
#### Problem
Booting from a snapshot further than the latest root in Blocktree causes a gap in the rooted slot iterator. This causes the rooted slot iterator to stop at these gaps.

#### Summary of Changes
Allow rooted slot iterator to jump past gaps caused by snapshots. Indicate a jump occurred by returning an `Option<SlotMeta>`

Fixes #

### CODE REVIEW

File: ledger/src/rooted_slot_iterator.rs [hunk 1]
```diff
@@ -134,4 +152,82 @@ mod tests {
...
+        let result: Vec<_> = RootedSlotIterator::new(3, &blocktree)
+            .unwrap()
+            .into_iter()
+            .map(|(slot, meta)| (slot, meta.is_some()))
+            .collect();
+        let expected = vec![(3, true), (10, false)];
+        assert_eq!(result, expected);
+
+        // Create one more post-skip slot at 11 with parent equal to 10
+        fill_blocktree_slot_with_ticks(&blocktree, ticks_per_slot, 11, 10, Hash::default());
```
Comment (CriesofCarrots) [inline ledger/src/rooted_slot_iterator.rs]: Needs `blocktree.set_roots(&[10]).unwrap();`?

Comment (carllin) [reply ledger/src/rooted_slot_iterator.rs]: yup, thanks!

**Comment by mvines** [comment]
@CriesofCarrots @carllin - what was the issue you encountered, if any, that precipitated this PR?  I'm trying to decide if we need a backport to v0.22

**Comment by CriesofCarrots** [comment]
@mvines , the issue this addresses is truncated data when there are gaps in the rooted slot chain: eg. a validator runs for a while, then goes offline, then catches back up from a snapshot (retaining its original ledger) and starts processing new blocks. The blocks both before and after the gap are legitimate roots, but this iterator would only recognize those on one side of the gap or the other at a time.

The `getBlockTime` method currently uses the RootedSlotIterator to find entries to amalgamate timestamp votes. If the RootedSlotIterator stops at gaps, that means the `getBlockTime` method would always return null for blocks after such gaps, even if the validator has the timestamp data to work with.
Similarly, `getConfirmedBlocks` would stop at a gap.

I think that it is probably worth backporting this to v0.22 in order to make those methods as robust as possible, although this will not affect nodes that validate straight through.

I'll let @carllin speak to details from other use-cases.

**Pull request merged**